source/Weather.cpp

```diff
@@ -70,12 +72,14 @@ const Point &Weather::Origin() const
 
 
 // Create any environmental effects and decrease the lifetime of this weather.
-void Weather::Step(vector<Visual> &visuals)
+void Weather::Step(vector<Visual> &visuals, const Point *effectCenter)
 {
 	// Environmental effects are created by choosing a random angle and distance from
 	// their origin, then creating the effect there.
 	double minRange = hazard->MinRange();
-	double maxRange = hazard->MaxRange();
+	// If it is systemwide, only draw around the flagship.
```

Comment:
I still feel that if we are talking about a system-wide (infinite) effect, then it feels more appropriate to generate the visible effects only on the visible screen-viewport (not in a circle range around the flagship, but in a rectangular area that has a size of `(screen::Width() + 2* effect-diameter) * Zoom()` by `(Screen::Height() + 2* effect-diameter) * Zoom()` and that is centered on the screen-viewport-center.

If we could accomplish this, then we would also avoid generating the effects that are in range in the circle but not actually visible on the viewport.

Reply:
Then what if the player is moving quickly or quickly zooms out? The fact that effects are only being generated within the bounds of the screen becomes immediately obvious.

Reply:
That is a good point, in that case we should indeed _generate_ the effects for the maximum zoom.

Reply:
Yeah, as I said too higher:
> (caus otherwise zooming out as the hazard ends means you notice it wasn't actually visually system-wide)

So that's why I did it this way, but I think what you're proposing is a bit overkill, like generating it in the exact square box is pushing it, considering the code generating a hazard is a circle I don't want to add an exception to win a few pixels either. 
You suggested on the base code btw and not the code I added for some reason, which is : 
`double maxRange = hazard->SystemWide() ? 2. * (Screen::Width() > Screen::Height() ? Screen::Width() : Screen::Height()) : hazard->MaxRange();`
It is needed to multiply it by two to account for max zoom (or maybe its because of something else, like pixels not directly translating to distance idk).

Also it best not to be exactly on top of the flagship, otherwise when the effect stops generating but the particles are still moving around you can see the square shape by moving even a bit, a circle shape is much less noticeable, I test it and it doesn't feel weird.